docs/source/conf.py

```diff
@@ -194,6 +194,32 @@ def setup(app):
     for css_file in html_css_files:
         add_css(css_file)
 
+# From PyTorch 1.5, we now use autogenerated files to document classes and
+# functions. This breaks older references since 
+# https://docs.pytorch.org/torch.html#torch.flip
+# moved to 
+# https://docs.pytorch.org/torch/generated/torchflip.html
+# which breaks older links from blog posts, stack overflow answers and more.
+# To mitigate that, we add an id="torch.flip" in an appropriated place
+# in torch.html by overriding the visit_reference method of html writers.
+# Someday this can be removed, once the old links fade away
+
+from sphinx.writers import html, html5
+
+def replace(Klass):
+    old_call = Klass.visit_reference
+
+    def visit_reference(self, node):
+        if 'refuri' in node and 'generated' in node.get('refuri'):
+            ref = node.get('refuri')
+            ref_anchor = ref.split('#')
+            if len(ref_anchor) > 1:
+                self.body.append('<p id="{}"/p>'.format(ref_anchor[1]))
```

Comment:
The empty `p` tag might introduce an empty line in some cases: 

| Before | After |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/6421097/83099458-0abc0980-a073-11ea-8cb6-b9348bd65d7f.png) | ![image](https://user-images.githubusercontent.com/6421097/83099436-fe37b100-a072-11ea-8c46-9e0c59767563.png) | 
| ![image](https://user-images.githubusercontent.com/6421097/83099498-2cb58c00-a073-11ea-90bf-689a1b3d9485.png) | ![image](https://user-images.githubusercontent.com/6421097/83099514-3b9c3e80-a073-11ea-916e-41843cb6b678.png) |

Reply:
The tag should not be emitted here at all. Fixed with more hacks to filter out other references and use only the one that is relevant

Reply:
As a minor side note here, we don’t put `p` after `/` for self closing tags, so it should be something like `<p id="..." />`

Reply:
fixed, thanks